structured/privilege.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package structured
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/cockroachdb/cockroach/security"
+	"github.com/cockroachdb/cockroach/sql/privilege"
+)
+
+// rawToPrivilegeList converts a set of privileges
+// into a PrivilegeList.
+func rawToPrivilegeList(m uint32) privilege.List {
+	ret := privilege.List{}
+	for _, p := range privilege.ByValue {
+		if m&(1<<p) != 0 {
+			ret = append(ret, p)
+		}
+	}
+
+	return ret
+}
```

Comment:
we can have it do all of it, but then that should include the descriptor manipulation as well, meaning depending on structured (eventually sql). The privilege package conveniently has no dependencies other than util/.